### PR TITLE
Feature/footer

### PR DESCRIPTION
**Pull request by PradeepRajput07**
Please review

### CODE REVIEW

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+import com.taylorsuniversity.models.bean.FooterModelBean;
+import com.taylorsuniversity.models.bean.FooterModelLinksBean;
+import com.taylorsuniversity.utils.CoreUtils;
+
+@Model(adaptables = SlingHttpServletRequest.class)
+public class FooterModel {
+	Logger LOGGER = LoggerFactory.getLogger(FooterModelBean.class);
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: FooterModelBaen.class?

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+	@PostConstruct
+	protected void init() {
+		socialProps();
+		footerLinkProps();
+	}
+
+	/*
+	 * Description: This method handles footer-links tab configuration
+	 */
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: Remove this doc.

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	private void footerLinkProps() {
+		
+		if (ArrayUtils.isNotEmpty(footerLinks)) {
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: Apart from empty check. Also check for null.

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+			this.footerLinksMap = new LinkedHashMap<String, List<FooterModelLinksBean>>();
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: use diamond operator.

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+	 *
+	 * @return the current year
+	 **/
+	public int getCurrentYear() {
+		Calendar cal = Calendar.getInstance();
+		LOGGER.debug("Current Year is : {} ", cal.get(Calendar.YEAR));
+		return cal.get(Calendar.YEAR);
+	}
+	
+	public List<FooterModelBean> getSocialItems() {
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: Add the javadoc.

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+		return socialItems;
+	}
+
+	public Map<String, List<FooterModelLinksBean>> getFooterLinksMap() {
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: Add the javadoc.

File: core/src/main/java/com/taylorsuniversity/models/bean/FooterModelBean.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+	private String socialText;
+	
+	public FooterModelBean(String socialIcon, String socialLink, String socialText) {
+		super();
+		this.socialIcon = socialIcon;
+		this.socialLink = socialLink;
+		this.socialText = socialText;
+	}
+	
+	public String getSocialIcon() {
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/bean/FooterModelBean.java]: Add the javadoc.

File: core/src/main/java/com/taylorsuniversity/models/bean/FooterModelLinksBean.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+	private String pageLinkItem;
+	private String pageNameItem;
+
+	public FooterModelLinksBean(String pageLink, String pageName) {
+		super();
+		this.pageLinkItem = pageLink;
+		this.pageNameItem = pageName;
+	}
+	
+	public String getPageLinkItem() {
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/bean/FooterModelLinksBean.java]: Add the javadoc.

File: ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/homepage/content.html [hunk 9]
```diff
@@ -1 +1,11 @@
-<div data-sly-resource="${'par' @ resourceType='wcm/foundation/components/responsivegrid'}"></div>
\ No newline at end of file
+<div class="btn-group sticky-left-buttons minimized d-none d-lg-block">
```
Comment (bsinghal1) [inline ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/homepage/content.html]: Why have you changed this file. This change is not required. Remove this file from the commit.

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+			}
+		}
+	}
+
+	/*
+	 * Description: This method handles social-links tab configuration
+	 */
+	private void socialProps() {
+		
+		if (ArrayUtils.isNotEmpty(social)) {
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: Check for null.

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+			socialItems = new ArrayList<>();
+			for (String si : social) {
+				try {
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: put some loggers.

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+					e.printStackTrace();
+				}
+				
+			}
+		}
+	}
+
+	/*
+	 * Description: This method handles social-links tab configuration
+	 */
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: remove this javadoc.

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 1 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: fixed

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 2 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: Done

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 3 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: done

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 4 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: done

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 12 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: done

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 10 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: done

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 5 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: Done

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 6 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: Done

File: core/src/main/java/com/taylorsuniversity/models/bean/FooterModelBean.java (hunk 7 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/bean/FooterModelBean.java]: done

File: core/src/main/java/com/taylorsuniversity/models/bean/FooterModelLinksBean.java (hunk 8 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/bean/FooterModelLinksBean.java]: done

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 11 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: done

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+import com.taylorsuniversity.models.bean.FooterModelBean;
+import com.taylorsuniversity.models.bean.FooterModelLinksBean;
+import com.taylorsuniversity.utils.CoreUtils;
+
+/**
+ * @author prajput
+ *
+ */
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: Change the comment to something like as below:-

/**
 * This class is used to retrieve the properties configured in the header component of the home-page
 * template.
 */

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+@SuppressWarnings("deprecation")
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: why is this required?

File: core/src/main/java/com/taylorsuniversity/models/bean/FooterModelBean.java [hunk 15]
```diff
@@ -0,0 +1,39 @@
+package com.taylorsuniversity.models.bean;
+
+/**
+ * @author prajput
+ *
+ */
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/bean/FooterModelBean.java]: Change the comment.

File: core/src/main/java/com/taylorsuniversity/models/bean/FooterModelLinksBean.java [hunk 16]
```diff
@@ -0,0 +1,30 @@
+package com.taylorsuniversity.models.bean;
+
+/**
+ * @author prajput
+ *
+ */
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/bean/FooterModelLinksBean.java]: Change the comment.

File: ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/homepage/content.html [hunk 17]
```diff
@@ -1 +1,11 @@
-<div data-sly-resource="${'par' @ resourceType='wcm/foundation/components/responsivegrid'}"></div>
\ No newline at end of file
+<div data-sly-resource="${'par' @ resourceType='wcm/foundation/components/responsivegrid'}"></div> 
+<div class="btn-group sticky-left-buttons minimized d-none d-lg-block">
```
Comment (bsinghal1) [inline ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/homepage/content.html]: Why this change is required? Remove this file from the commit.

File: ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/basepage/footer.html [hunk 18]
```diff
@@ -19,5 +19,5 @@
          data-sly-call="${clientLib.js @ categories=clientlibCategories}"></sly>
     <sly data-sly-include="customfooterlibs.html"></sly>
     <sly data-sly-resource="${'cloudservices' @ resourceType='cq/cloudserviceconfigs/components/servicecomponents'}"></sly>
-    <sly data-sly-include="/apps/taylorsuniversity/components/content/footer/content/content.html"></sly>
+    <div data-sly-resource="${ @path='footer', resourceType='taylorsuniversity/components/content/footer', wcmmode=mode}"></div>
```
Comment (bsinghal1) [inline ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/basepage/footer.html]: move this to the body.html for the time being.

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 13 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: done

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 14 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: Because the JSONObject API is deprecated. So using this method is giving warnings. Hence supressing.

File: core/src/main/java/com/taylorsuniversity/models/bean/FooterModelBean.java (hunk 15 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/bean/FooterModelBean.java]: Done

File: core/src/main/java/com/taylorsuniversity/models/bean/FooterModelLinksBean.java (hunk 16 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/bean/FooterModelLinksBean.java]: Done

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 19]
```diff
@@ -0,0 +1,168 @@
...
+						for (FooterModelLinkPathBean li: pages) {
+							
+							if(StringUtils.isNotBlank(li.getLinkPath())) {
+								
+								link = li.getLinkPath();
+								LOGGER.debug("Link is : {}" + link);
+								
+								if(CoreUtils.isInternalLink(link)) {
+									linkName = pageManager.getPage(link).getTitle();
+								}
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: null check for pageManager.getPage(link).

File: core/src/main/java/com/taylorsuniversity/models/bean/FooterModelSocialBean.java [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+	 */
+	public void setSocialIcon(String socialIcon) {
+		this.socialIcon = socialIcon;
+	}
+
+
+	/**
+	 * @param socialLink
+	 */
+	public void setSocialLink(String socialLink) {
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/bean/FooterModelSocialBean.java]: do we require setter method in it?

File: core/src/main/java/com/taylorsuniversity/models/bean/FooterModelLinkPathBean.java [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+ */
+public class FooterModelLinkPathBean {
+	
+	private String linkPath;
+
+	public String getLinkPath() {
+		return linkPath;
+	}
+
+	public void setLinkPath(String linkPath) {
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/bean/FooterModelLinkPathBean.java]: setter method?

File: core/src/main/java/com/taylorsuniversity/models/bean/FooterModelLinksBean.java [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+
+public class FooterModelLinksBean {
+	
+	private String footerLinksTitle;
+	private List<FooterModelLinkPathBean> footerLinksPages;
+	
+	public String getFooterLinksTitle() {
+		return footerLinksTitle;
+	}
+	public void setFooterLinksTitle(String footerLinksTitle) {
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/bean/FooterModelLinksBean.java]: setter method?

File: ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/basepage/body.html [hunk 23]
```diff
@@ -41,6 +41,7 @@
  <!-- Placeholder for header component -->
 <sly data-sly-include="content.html"></sly>
 <!-- Placeholder for footer component -->
+<div data-sly-resource="${ @path='footer', resourceType='taylorsuniversity/components/content/footer', wcmmode=mode}"></div>
```
Comment (bsinghal1) [inline ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/basepage/body.html]: Change this line to the following:-

<sly data-sly-test.templateName="${pageProperties.cq:template}">
	<div data-sly-test.mode="${templateName == '/apps/taylorsuniversity/templates/homepage' ? 'edit' : 'disabled'}">
		<section class="container-fluid">
			<div data-sly-resource="${ @path='footer', resourceType='taylorsuniversity/components/content/footer', wcmmode=mode}"></div>
		</section>
	</div>
</sly>

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 19 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: done

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 24]
```diff
@@ -0,0 +1,171 @@
...
+	}
+
+	/**
+	 * Reads the authored footer links properties and adds to bean
+	 * 
+	 */
+	private void footerLinkProps() {
+						
+		footerLinks = rootPage.getContentResource("footer").getValueMap().get("footerLinks",String[].class);
+		LOGGER.debug("FooterLinks value is: {} " + footerLinks);
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: null checks for "rootPage" and subsequent invocations

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java [hunk 25]
```diff
@@ -0,0 +1,171 @@
...
+			}
+		}
+	}
+
+	/**
+	 * Reads the authored social properties and adds to bean
+	 * 
+	 */
+	private void socialProps() {
+		social = rootPage.getContentResource("footer").getValueMap().get("social",String[].class);
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: do the same null checks here as well.

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 24 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: fixed

File: core/src/main/java/com/taylorsuniversity/models/FooterModel.java (hunk 25 above)
Comment (PradeepRajput07) [reply core/src/main/java/com/taylorsuniversity/models/FooterModel.java]: fixed
